File: TMessagesProj/src/main/java/org/telegram/telekill/TeleKillConfig.java
package org.telegram.telekill;

import android.content.Context;
import android.content.SharedPreferences;

public class TeleKillConfig {

    private static final String PREFS_NAME = "telekill_config";
    private static SharedPreferences prefs;

    public static final int GHOST_TYPE_ALL = 0;
    public static final int GHOST_TYPE_CONTACTS_ONLY = 1;
    public static final int GHOST_TYPE_EXCEPT_FAVORITES = 2;
    public static final int GHOST_TYPE_GROUPS_ONLY = 3;

    public static void init(Context context) {
        prefs = context.getApplicationContext().getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE);
    }

    // Ghost Mode
    public static boolean isGhostModeEnabled() {
        return prefs.getBoolean("ghost_mode", false);
    }
    public static void setGhostModeEnabled(boolean value) {
        prefs.edit().putBoolean("ghost_mode", value).apply();
    }

    public static boolean isGhostHideReadStatus() {
        return prefs.getBoolean("ghost_hide_read", true);
    }
    public static void setGhostHideReadStatus(boolean value) {
        prefs.edit().putBoolean("ghost_hide_read", value).apply();
    }

    public static boolean isGhostHideTyping() {
        return prefs.getBoolean("ghost_hide_typing", true);
    }
    public static void setGhostHideTyping(boolean value) {
        prefs.edit().putBoolean("ghost_hide_typing", value).apply();
    }

    public static boolean isGhostHideOnline() {
        return prefs.getBoolean("ghost_hide_online", true);
    }
    public static void setGhostHideOnline(boolean value) {
        prefs.edit().putBoolean("ghost_hide_online", value).apply();
    }

    public static int getGhostModeType() {
        return prefs.getInt("ghost_mode_type", GHOST_TYPE_ALL);
    }
    public static void setGhostModeType(int value) {
        prefs.edit().putInt("ghost_mode_type", value).apply();
    }

    // Proxy
    public static boolean isAutoProxyEnabled() {
        return prefs.getBoolean("auto_proxy", true);
    }
    public static void setAutoProxyEnabled(boolean value) {
        prefs.edit().putBoolean("auto_proxy", value).apply();
    }

    public static boolean isProxySetupDone() {
        return prefs.getBoolean("proxy_setup_done", false);
    }
    public static void setProxySetupDone(boolean value) {
        prefs.edit().putBoolean("proxy_setup_done", value).apply();
    }
}

File: TMessagesProj/src/main/java/org/telegram/telekill/TeleKillConfig.kt
package org.telegram.telekill

import android.content.Context
import android.content.SharedPreferences

object TeleKillConfig {

    private const val PREFS_NAME = "telekill_config"

    private lateinit var prefs: SharedPreferences

    fun init(context: Context) {
        prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
    }

    // Ghost Mode
    var ghostModeEnabled: Boolean
        get() = prefs.getBoolean("ghost_mode", false)
        set(value) = prefs.edit().putBoolean("ghost_mode", value).apply()

    var ghostHideReadStatus: Boolean
        get() = prefs.getBoolean("ghost_hide_read", true)
        set(value) = prefs.edit().putBoolean("ghost_hide_read", value).apply()

    var ghostHideTyping: Boolean
        get() = prefs.getBoolean("ghost_hide_typing", true)
        set(value) = prefs.edit().putBoolean("ghost_hide_typing", value).apply()

    var ghostHideOnline: Boolean
        get() = prefs.getBoolean("ghost_hide_online", true)
        set(value) = prefs.edit().putBoolean("ghost_hide_online", value).apply()

    // Ghost mode types
    var ghostModeType: Int
        get() = prefs.getInt("ghost_mode_type", GHOST_TYPE_ALL)
        set(value) = prefs.edit().putInt("ghost_mode_type", value).apply()

    // Proxy
    var autoProxyEnabled: Boolean
        get() = prefs.getBoolean("auto_proxy", true)
        set(value) = prefs.edit().putBoolean("auto_proxy", value).apply()

    var proxySetupDone: Boolean
        get() = prefs.getBoolean("proxy_setup_done", false)
        set(value) = prefs.edit().putBoolean("proxy_setup_done", value).apply()

    companion object {
        const val GHOST_TYPE_ALL = 0           // для всех
        const val GHOST_TYPE_CONTACTS_ONLY = 1 // только для контактов
        const val GHOST_TYPE_EXCEPT_FAVORITES = 2 // для всех кроме избранных
        const val GHOST_TYPE_GROUPS_ONLY = 3   // только в группах
    }
}
